### PR TITLE
CNS-794-verify-policy-geolocation-on-tx-verification

### PR DESCRIPTION
**Pull request by Yaroms**

### CODE REVIEW

File: x/projects/types/message_set_subscription_policy.go [hunk 1]
```diff
@@ -46,5 +46,9 @@ func (msg *MsgSetSubscriptionPolicy) ValidateBasic() error {
 		return sdkerrors.Wrapf(legacyerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
 
+	if _, ok := planstypes.Geolocation_name[msg.Policy.GeolocationProfile]; !ok {
+		return sdkerrors.Wrapf(legacyerrors.ErrInvalidType, "invalid geolocationprofile")
```
Comment (oren-lava) [inline x/projects/types/message_set_subscription_policy.go:50]: same

File: x/projects/types/message_set_admin_policy.go [hunk 2]
```diff
@@ -46,5 +46,9 @@ func (msg *MsgSetPolicy) ValidateBasic() error {
 		return sdkerrors.Wrapf(legacyerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
 
+	if _, ok := planstypes.Geolocation_name[msg.Policy.GeolocationProfile]; !ok {
```
Comment (oren-lava) [inline x/projects/types/message_set_admin_policy.go:49]: put both verifications in `x/plans/types/policy.go`

File: x/projects/types/message_set_subscription_policy.go (hunk 1 above)
Comment (Yaroms) [reply x/projects/types/message_set_subscription_policy.go]: https://github.com/lavanet/lava/pull/1239/commits/76b61f97d1d5ff912b9079f09a1c6ee1f2944e63

File: x/projects/types/message_set_admin_policy.go [hunk 3]
```diff
@@ -46,5 +46,9 @@ func (msg *MsgSetPolicy) ValidateBasic() error {
...
+		return sdkerrors.Wrapf(legacyerrors.ErrInvalidType, "invalid geolocationprofile")
```
[the remark this answers is not in the document]
Comment (Yaroms) [reply x/projects/types/message_set_admin_policy.go]: https://github.com/lavanet/lava/pull/1239/commits/76b61f97d1d5ff912b9079f09a1c6ee1f2944e63

**Review by oren-lava — APPROVED** [review]

**Pull request merged**